src/releasing/quattorpoddoc.py

```diff
@@ -13,278 +16,384 @@
 from vsc.utils.generaloption import simple_option
 from vsc.utils import fancylogger
 from vsc.utils.run import run_asyncloop
+import tempfile
+from lxml import etree
+import shutil
 
 MAILREGEX = re.compile(("([a-z0-9!#$%&'*+\/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+\/=?^_`"
                         "{|}~-]+)*(@|\sat\s)(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?(\.|"
                         "\sdot\s))+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)"))
 PATHREGEX = re.compile(r'(\s+)((?:/[\w{}]+)+\.?\w*)(\s*)')
 EXAMPLEMAILS = ["example", "username", "system.admin"]
-LOGGER = fancylogger.getLogger()
-SUBDIR = "components"
+logger = fancylogger.getLogger()
 DOCDIR = "docs"
 
+REPOMAP = {
+    "configuration-modules-core": {"sitesubdir": "components",
+                                   "target": "/NCM/Component/"},
+    "CAF": {"sitesubdir": "CAF",
+            "target": "/CAF/"},
+    "CCM": {"sitesubdir": "CCM",
+            "target": "EDG/WP4/CCM/"},
+    }
+
 
-def mavencleancompile(modules_location):
+def mavencleancompile(repo_location, repository):
     """
     Executes mvn clean and mvn compile in the given modules_location.
     """
+    repoloc = os.path.join(repo_location, repository)
+    logger.info("Doing maven clean in %s." % repoloc)
+    output = run_asyncloop("mvn clean", startpath=repoloc)
+    logger.debug(output)
 
-    LOGGER.info("Doing maven clean in %s." % modules_location)
-    output = run_asyncloop("mvn clean", startpath=modules_location)
-    LOGGER.debug(output)
-
-    LOGGER.info("Doing maven compile in %s." % modules_location)
-    output = run_asyncloop("mvn compile", startpath=modules_location)
-    LOGGER.debug(output)
+    logger.info("Doing maven compile in %s." % repoloc)
+    output = run_asyncloop("mvn compile", startpath=repoloc)
+    logger.debug(output)
 
 
-def generatemds(pods, location):
+def generatemds(repo, sources, location):
     """
     Takes a list of components with podfiles and generates a md file for it.
     """
-    LOGGER.info("Generating md files.")
-    counter = 0
+    logger.info("Generating md files.")
     mdfiles = []
 
-    comppath = os.path.join(location, DOCDIR, SUBDIR)
+    comppath = os.path.join(location, DOCDIR, REPOMAP[repo]['sitesubdir'])
     if not os.path.exists(comppath):
         os.makedirs(comppath)
 
-    for component in sorted(pods):
-        for pod in pods[component]:
-            component = component.replace('ncm-', '')
-            title = os.path.splitext(os.path.basename(pod))[0]
-            if component != title:
-                title = component+'::'+title
-            mdfile = os.path.join(comppath, '%s.md' % title)
-            convertpodtomarkdown(pod, mdfile)
-            counter += 1
+    for source in sources:
+        if source.endswith(".pan"):
```

Comment:
add a `endswith('.tpl')` which should log a warning and then continue like `.pan`